examples/main/node-psbt/index.ts

```diff
@@ -0,0 +1,192 @@
+// run `node --import=tsx index.ts` in the terminal
+
+import {
+  BestRouteRequest,
+  CreateTransactionRequest,
+  RangoClient,
+  TransactionStatus,
+  TransactionType,
+} from 'rango-sdk'
+import { findToken } from '../shared/utils/meta.js'
+import {
+  logMeta,
+  logSelectedTokens,
+  logWallet,
+  logTransactionHash,
+  logRoutes,
+  logStepStatus,
+  logConfirmedRoute,
+} from '../shared/utils/logger.js'
+import { setTimeout } from 'timers/promises'
+
+import * as bitcoin from 'bitcoinjs-lib'
+import { ECPairFactory } from 'ecpair'
+import * as tinysecp from 'tiny-secp256k1'
+
+// Configuration parameters
+const BTC_RPC_URL = 'https://go.getblock.io/f37bad28a991436483c0a3679a3acbee'
+const BITCOIN_WIF = 'YOUR_BITCOIN_PRIVATE_KEY'
+const EVM_ADDRESS = 'YOUR_EVM_ADDRESS'
+const API_KEY = 'c6381a79-2817-4602-83bf-6a641a409e32'
+const NETWORK = bitcoin.networks.bitcoin // Use testnet: bitcoin.networks.testnet
+const MAX_ROUTES = 1 // Limit to one route
+const STEP_INDEX = 1 // Execute the first step
+
+// Example request values
+const SOURCE_CHAIN = 'BTC'
+const SOURCE_TOKEN_ADDR = null
+const TARGET_CHAIN = 'ARBITRUM'
+const TARGET_TOKEN_ADDR = '0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f'
+const AMOUNT = '0.0002'
+const SLIPPAGE = '1.0'
+
+async function processSwapRoutes() {
+  try {
+    // Step 1: Load and validate wallet
+    const ECPair = ECPairFactory(tinysecp)
+    const keyPair = ECPair.fromWIF(BITCOIN_WIF, NETWORK)
+    const { address } = bitcoin.payments.p2wpkh({
+      pubkey: Buffer.from(keyPair.publicKey),
+      network: NETWORK,
+    })
+
+    if (!address) throw new Error('Failed to derive wallet address')
+    logWallet(address)
+
+    // Step 2: Initialize client and fetch metadata
+    const rango = new RangoClient(API_KEY)
+    const metadata = await rango.getAllMetadata()
+    logMeta(metadata)
+
+    // Step 3: Select tokens for routing
+    const sourceToken = findToken(
+      metadata.tokens,
+      SOURCE_CHAIN,
+      SOURCE_TOKEN_ADDR
+    )
+    const targetToken = findToken(
+      metadata.tokens,
+      TARGET_CHAIN,
+      TARGET_TOKEN_ADDR
+    )
+    logSelectedTokens(sourceToken, targetToken)
+
+    // Step 4: Request possible routes
+    const routeRequest: BestRouteRequest = {
+      from: sourceToken,
+      to: targetToken,
+      amount: AMOUNT,
+      slippage: SLIPPAGE,
+      maxLength: MAX_ROUTES,
```

Comment:
`maxLength` is specifying the maximum number of steps.
```suggestion
      maxLength: MAX_STEPS,
```

Reply:
Thanks for catching that.
I fixed it.